packages/core/src/resolvers/arguments.ts

```diff
@@ -39,44 +42,64 @@ const resolveQueryArgs = async (
 };
 
 /**
- * Deeply resolve a filter argument into a set of nodes with relations and not-yet resolved fields.
- *
- * Immutably filters and returns a new array.
+ * Deeply resolves a filter argument as a subquery, and returns a set of content node IDs that satisfy the filter.
  *
  * @param filter the filter argument
- * @param nodes the array of nodes to filter
  */
 export const resolveFilter = async (
   filter: Record<string, any>,
-  nodes: any[],
-  type: string,
-  schemaComposer: SchemaComposer
-) => {
-  // construct custom resolver of nodes to build temp list
+  config: FlatbreadConfig,
+  options: ResolveQueryArgsOptions
+): Promise<(string | number)[]> => {
+  // Seperate the filter into its parts:
+  //  - the path leading to the field we want to compare
+  //  - the comparator expression.
   const filterSetManifest = generateFilterSetManifest(filter);
-  console.log('filter manifest', filterSetManifest);
+
+  // Run Flatbread as a function to execute a subquery
+  const flatbread = new FlatbreadProvider(config);
+
+  let filterToQuery = [];
 
   for (const filter of filterSetManifest) {
-    for (const field of filter.path) {
-      const objectTC = schemaComposer.getOTC(type);
-      let fieldTC = objectTC.hasField(field) && objectTC.getField(field);
+    let graphQLFieldAccessor = '';
 
-      if (fieldTC instanceof ThunkComposer) {
-        fieldTC = fieldTC.getUnwrappedTC();
+    for (let i = 0; i < filter.path.length; i++) {
+      const field = filter.path[i];
+      const lastFieldIndex = filter.path.length - 1;
+
+      // Build a partial GraphQL query's field shape
+      if (i === lastFieldIndex && filter.path.length === 1) {
+        graphQLFieldAccessor += `${field}`;
+      } else if (i !== lastFieldIndex) {
+        graphQLFieldAccessor += `${field} {`;
+      } else {
+        graphQLFieldAccessor += `${field} ${[lastFieldIndex]
+          .map(() => '}')
```

Comment:
can you add asome comments here? not clear why you're matching `{` and `}` across branches

Reply:
Good call, I extracted it into its own function + added comments with an example